src/rust/engine/src/context.rs

```diff
@@ -94,6 +95,8 @@ impl Core {
         None => Box::new(process_execution::local::CommandRunner::new(
           store.clone(),
           fs_pool.clone(),
+          work_dir.clone().to_path_buf(),
```

Comment:
I have a personal style preference that if a function needs to own some data, it should take an owned version of it, rather than taking a reference and cloning it, because that's more upfront about the cost. So personally I would change this function signature to take a `PathBuf` instead of a `&Path`, rather than doing the cloning here. In this case, it avoids the clone entirely, because the caller is actually doing: `work_dir_buf.to_os_string().as_ref()`, so rather than doing `work_dir_buf.to_os_string().as_ref().clone().to_path_buf()` we should end up just doing `PathBuf::from(work_dir_buf.to_os_string())` and being happy.

(Also, I think `work_dir.clone().to_path_buf()` could be re-written as just `work_dir.to_path_buf()`, but the above is probably better :))

Reply:
not seeing the `&Path` you're referring to here?

but went with `work_dir.to_path_buf()` instead.

Reply:
Aah, I meant the actual `work_dir` arg; if you change its type to `PathBuf`, and have `scheduler_create` call `PathBuf::from(work_dir.to_os_string())` instead of `work_dir_buf.to_os_string().as_ref()`, you can save a copy, and express intent a little more cleanly